### PR TITLE
java.d: prevent integer overflow when updateEvery > 2s

### PR DESCRIPTION
**Pull request by illes**

### CODE REVIEW

File: java.d/src/main/java/org/firehol/netdata/utils/AlignToTimeIntervalService.java [hunk 1]
```diff
@@ -18,16 +18,19 @@
...
 
-	private long intervalInNSec;
+	private final long intervalInNSec;
 	private long lastTimestamp;
 
-	public AlignToTimeIntervalService(long intervalInNSec) {
-		this.intervalInNSec = intervalInNSec;
+	public AlignToTimeIntervalService(long interval, TimeUnit unit) {
+		if (interval <= 0)
+			throw new IllegalArgumentException("interval cannot be negative: " + interval);
```
Comment (simartn) [inline java.d/src/main/java/org/firehol/netdata/utils/AlignToTimeIntervalService.java]: Error message does not fit edge case 0:
interval cannot be negative: 0

**Review by simartn — CHANGES_REQUESTED** [review]
Thank you for this commit. Can you also provide a test case which would fail with the previous implementation to prevent rolling it back once again?

**Comment by illes** [comment]
Gotcha, rebased on the test-case 5d182bb

**Review by simartn — APPROVED** [review]
